### ISSUE TITLE
gettext fails to build on Bash on Ubuntu on Windows (Ubuntu 14.04)

### ISSUE THREAD

**Issue by mshindal** [mxe/mxe#1518]
https://gist.github.com/mshindal/f4e02701f32dcb170bd4070c73f1a8e2

**Comment by mshindal** [comment]
That didn't seem to work. Just to clarify, you're saying I should run `ln -s ../automake.mk plugins/native/trusty/automake.mk` inside of my cloned `mxe` directory?

**Comment by darealshinji** [comment]
Yes.

**Comment by mshindal** [comment]
I get this error: 
```make: *** No rule to make target `/mnt/d/Code/mxe/usr/x86_64-unknown-linux-gnu/installed/autoconf', needed by `/mnt/d/Code/mxe/usr/x86_64-unknown-linux-gnu/installed/automake'.  Stop.```

**Comment by darealshinji** [comment]
Ah, dependencies. Create links for `autoconf.mk` and `m4.mk` too and try it again.

**Comment by tonytheodore** [comment]
I'm not sure manual symlinks are the best way to go (you want `libtool` also), either of:

```
make OS_SHORT_NAME=wheezy ...
```
or
```
make libtool MXE_PLUGIN_DIRS=plugins/native MXE_TARGETS=`ext/config.guess`
```

will build a working autotools setup.

The core issue is why `aclocal` is called at all? We don't modify any of the base files so it shouldn't be needed for this package.

**Comment by darealshinji** [comment]
My guess is that Ubuntu on Windows 10 might behave different than a real Ubuntu system. Last time I tried to compile something there it didn't work at all.

**Comment by tonytheodore** [comment]
Good point @darealshinji, I haven't used it before - I wonder if there's some weirdness with timestamps on the files extracted from the tarball?

**Comment by starius** [comment]
@mshindal 
What is the output of `lsb_release -sc` and `uname -s`?

**Comment by mshindal** [comment]
`trusty` and `Linux`.

**Comment by starius** [comment]
@tonytheodore
Why not to build these native packages on trusty to make it work out-of-box?

**Comment by darealshinji** [comment]
It seems to be working now.

**Comment by Saniq** [comment]
@darealshinji it might be fixed due to the newest version of _Bash on Ubuntu on Windows_ coming with Ubuntu 16.04 which in turn has automake-1.15 in its package source. But the underlying issue still seems to persist (at least for me on Windows 10 Creators Update (1703 / 15063.332)).

Other packages are now asking for aclocal-1.14 or even aclocal-1.13.

I am a novice when it comes to autotools but shouldn't it be possible to regenerate the build files with every newer version of those?

**Comment by darealshinji** [comment]
> I am a novice when it comes to autotools but shouldn't it be possible to regenerate the build files with every newer version of those?

Yes, `autoreconf -if` should to that. Can you point out which packages this happened to you? I'd like to see if I can re-create that issue.

**Comment by Saniq** [comment]
This happend to me with _libmng_ requiring `aclocal-1.13` and _libidn2_ requiring `aclocal-1.14`.

While I was not able to build _libmng_ at all, I got _libidn2_ to build with `automake-1.14` manually installed alongside the current version and [this](https://gist.github.com/Saniq/934659235a51a0caa1393f4b2aba6248) patch applied to the package.
The patch is needed since the patch currently in place only modifies `Makefile.in` which gets overwritten when the file is being regenerated.

**Comment by darealshinji** [comment]
I just finished building libmng for i686-w64-mingw32.static without problems. Where did you save your mxe directory to? I think trying to build something outside of the pseudo-root partition may cause issues (probably something with timestamps).

**Comment by Saniq** [comment]
> Where did you save your mxe directory to?

Yes, my bad. I used mxe from within the Windows filesystem. Building the mentioned packages under the pseudo-root partition works for me as well.

Has building from the Linux filesystem always worked or has it only started to work with recent Windows Insider Previews or the Creators Update ?

**Comment by darealshinji** [comment]
> Has building from the Linux filesystem always worked or has it only started to work with recent Windows Insider Previews or the Creators Update ?

Good question. The user who started this was building on a Windows filesystem which may have caused the issues in the first place.

**Comment by Saniq** [comment]
I would assume so too.
Would it be possible to detect wether a certain package tries to regenerate the build files with outdated versions of autotools and call `autoreconf -if` in that case? This would allow mxe to build when saved in the Windows filesystem.

In any case the issue can be considered fixed since mxe works fine from within the Linux filesystem.

**Comment by darealshinji** [comment]
Haven't checked it yet but using http://pkg.mxe.cc/ on `Bash on Windows` might be a good option.
